<img src="images/bannerugentdwengo.png" alt="Banner" width="400"/>

<div>
    <font color=#690027 markdown="1">
        <h1>TOEPASSING BIJ ABSOLUTE EN RELATIEVE FREQUENTIE</h1> 
    </font>
</div>

<div>
    <font color=#690027 markdown="1">
        <h2>RUIMTEMISSIES</h2> 
    </font>
</div>

In deze notebook gebruik je de dataset *Space Mission Launches*. Deze dataset wordt aangeboden in een csv-bestand `mission_launches.csv` in de map `data`. 

### Nodige modules importeren

In [1]:
import pandas as pd  # importeer pandas-module

<div>
    <font color=#690027 markdown="1">
        <h3>1. Absolute frequentie</h3> 
    </font>
</div>

### Opdracht 1.1

Lees de dataset in met Python. Verwijs ernaar met de variabele `ruimtemissies`.

In [ ]:
# dataset inladen


In [2]:
# voorbeeldcode
# dataset inladen
ruimtemissies = pd.read_csv("data/mission_launches.csv")    # dataset inlezen
ruimtemissies                                               # dataset laten zien

Unnamed: 0.1  Unnamed: 0 Organisation  \
0                0           0       SpaceX   
1                1           1         CASC   
2                2           2       SpaceX   
3                3           3    Roscosmos   
4                4           4          ULA   
...            ...         ...          ...   
4319          4319        4319      US Navy   
4320          4320        4320         AMBA   
4321          4321        4321      US Navy   
4322          4322        4322    RVSN USSR   
4323          4323        4323    RVSN USSR   

                                               Location  \
0            LC-39A, Kennedy Space Center, Florida, USA   
1     Site 9401 (SLS-2), Jiuquan Satellite Launch Ce...   
2                         Pad A, Boca Chica, Texas, USA   
3          Site 200/39, Baikonur Cosmodrome, Kazakhstan   
4              SLC-41, Cape Canaveral AFS, Florida, USA   
...                                                 ...   
4319           LC-18A, Cape Canaveral AFS, Florida, USA   
4320           LC-26A, Cape Canaveral AFS, Florida, USA   
4321           LC-18A, Cape Canaveral AFS, Florida, USA   
4322          Site 1/5, Baikonur Cosmodrome, Kazakhstan   
4323          Site 1/5, Baikonur Cosmodrome, Kazakhstan   

                            Date  \
0     Fri Aug 07, 2020 05:12 UTC   
1     Thu Aug 06, 2020 04:01 UTC   
2     Tue Aug 04, 2020 23:57 UTC   
3     Thu Jul 30, 2020 21:25 UTC   
4     Thu Jul 30, 2020 11:50 UTC   
...                          ...   
4319  Wed Feb 05, 1958 07:33 UTC   
4320  Sat Feb 01, 1958 03:48 UTC   
4321  Fri Dec 06, 1957 16:44 UTC   
4322  Sun Nov 03, 1957 02:30 UTC   
4323  Fri Oct 04, 1957 19:28 UTC   

                                            Detail  Rocket_Status  Price  \
0     Falcon 9 Block 5 | Starlink V1 L9 & BlackSky   StatusActive   50.0   
1              Long March 2D | Gaofen-9 04 & Q-SAT   StatusActive  29.75   
2               Starship Prototype | 150 Meter Hop   StatusActive    NaN   
3     Proton-M/Briz-M | Ekspress-80 & Ekspress-103   StatusActive   65.0   
4                       Atlas V 541 | Perseverance   StatusActive  145.0   
...                                            ...            ...    ...   
4319                     Vanguard | Vanguard TV3BU  StatusRetired    NaN   
4320                           Juno I | Explorer 1  StatusRetired    NaN   
4321                       Vanguard | Vanguard TV3  StatusRetired    NaN   
4322                    Sputnik 8K71PS | Sputnik-2  StatusRetired    NaN   
4323                    Sputnik 8K71PS | Sputnik-1  StatusRetired    NaN   

     Mission_Status  
0           Success  
1           Success  
2           Success  
3           Success  
4           Success  
...             ...  
4319        Failure  
4320        Success  
4321        Failure  
4322        Success  
4323        Success  

[4324 rows x 9 columns]

<div class="alert alert-box alert-success">
Deze dataset ziet er op het eerste zicht vrij ingewikkeld uit. Ze is afkomstig van <b>Kaggle</b>. Dat is een platform dat onder andere datasets gratis beschikbaar stelt. Je kan de bron van deze data zelf checken via <a href="https://www.kaggle.com/datasets/sefercanapaydn/mission-launches">https://www.kaggle.com/datasets/sefercanapaydn/mission-launches</a>. Daar vind je meer info over waar de data juist vandaan komt, en wat erin staat.    
</div> 

### Opdracht 1.2

-  Hoeveel ruimtemissies bevat de tabel?
-  Hoeveel kenmerken van ruimtemissies vind je terug?

Antwoord:

Antwoord: De dataset bevat 4324 rijen die elk gegevens bevatten over 1 ruimtemissie. Elk kenmerk van een ruimtemissie wordt in een aparte kolom weergegeven.<br>
De eerste twee kolommen lijken niets interessants te bevatten, voor mij toch niet interessant.
Er zijn dus 4324 ruimtemissies met elk 7 kenmerken, waaronder de naam van de missie.

### Opdracht 1.3

Je zal voor elke organisatie onderzoeken hoeveel ruimtemissies zij al hebben uitgevoerd.

-  De kolom `Organisation` toont voor elke ruimtemissie welke organisatie die uitgevoerd heeft. Dus maak alvast een variabele `organisaties` aan die naar die kolom verwijst.

In [ ]:
# organisaties


In [3]:
# voorbeeldcode
# organisaties
organisaties = ruimtemissies["Organisation"]  # kolom Organisation uit de dataset
organisaties  # kolom Organisation laten zien op het scherm

0          SpaceX
1            CASC
2          SpaceX
3       Roscosmos
4             ULA
          ...    
4319      US Navy
4320         AMBA
4321      US Navy
4322    RVSN USSR
4323    RVSN USSR
Name: Organisation, Length: 4324, dtype: object

- Hoeveel ruimtemisies heeft elke organisatie uitgevoerd?

In [ ]:
# aantal ruimtemissies per organisatie
# voorbeeldcode


In [4]:
# aantal ruimtemissies per organisatie
# voorbeeldcode
absolutefrequentietabel_organisaties = organisaties.value_counts()  # absolutefrequentietabel van organisaties
absolutefrequentietabel_organisaties  # absolutefrequentietabel laten zien op het scherm

RVSN USSR           1777
Arianespace          279
CASC                 251
General Dynamics     251
NASA                 203
VKS RF               201
US Air Force         161
ULA                  140
Boeing               136
Martin Marietta      114
SpaceX               100
MHI                   84
Northrop              83
Lockheed              79
ISRO                  76
Roscosmos             55
ILS                   46
Sea Launch            36
ISAS                  30
Kosmotras             22
US Navy               17
ISA                   13
Rocket Lab            13
Eurockot              13
ESA                   13
Blue Origin           12
IAI                   11
ExPace                10
ASI                    9
CNES                   8
AMBA                   8
MITT                   7
JAXA                   7
Land Launch            7
UT                     5
KCST                   5
CASIC                  5
Exos                   4
CECLES                 4
Arm??e de l'Air        4


<div class="alert alert-box alert-info">
Om te weten hoeveel ruimtemissies elke organisatie heeft uitgevoerd, stelde je de absolutefrequentiestabel op. Daarin is het aantal ruimtemisies per organisatie af te lezen.<br>
Om de absolutefrequentiestabel op te stellen, moet er geteld worden. 4324 waarden tellen zou voor een mens enkele uren werk zijn; voor een computer duurt dat enkele milliseconden.
</div>

Met de absolutefrequentietabel ter beschikking kan je uit de dataset, die op het eerste zicht ingewikkeld leek, al een hoop nuttige informatie afleiden.

<div class="alert alert-box alert-info">
Merk op dat 'data' en 'informatie' geen synoniemen zijn.
 </div>

<div>
    <font color=#690027 markdown="1">
        <h3>2. Informatie</h3> 
    </font>
</div>

### Opdracht 2.1

- Welke organisatie heeft al de meeste missies volbracht?

Antwoord:

Antwoord: RVSN USSR


- Hoeveel missies heeft Roscosmos al volbracht?

Antwoord:

Antwoord: 55

Uiteraard kan je dit aflezen uit de absolutefrequentietabel. Maar je kan dit ook opvragen a.d.h.v. code.

### Opdracht 2.2

- Schrijf code die met een varaiabele verwijst naar het aantal ruimtemissies door Boeing  en dat aantal laat zien op het scherm.

In [ ]:
# Boeing


In [5]:
# voorbeeldcode
# Boeing
absolute_frequentie_Boeing = absolutefrequentietabel_organisaties["Boeing"]  
absolute_frequentie_Boeing                                                   

136

### Opdracht 2.3
- Bereken de absolute frequentie van de waarde `SpaceX` en verwijs ernaar met de variabele `absolute_frequentie_SpaceX`. 
- Laat vervolgens de som van het aantal ruimtemissies dat door Boeing en door SpaceX werden georganiseerd, zien op het scherm. Voorzie hier een zinnetje bij.

In [6]:
# voorbeeldcode
# SpaceX
absolute_frequentie_SpaceX = absolutefrequentietabel_organisaties["SpaceX"]  

# totaal
som = absolute_frequentie_Boeing  + absolute_frequentie_SpaceX 
print("Het aantal missies van SpaceX en Boeing is samen", som)

Het aantal missies van SpaceX en Boeing is samen 236


### Opdracht 2.4
- Bereken de grootte van de *mission_launches* dataset en verwijs ernaar met de variabele `aantal_ruimtemissies`.

In [14]:
# voorbeeldcode
aantal_ruimtemissies = len(ruimtemissies)

- Hoeveel rijen bevat de dataset?

In [15]:
# voorbeeldcode
print(aantal_ruimtemissies)

4324


Antwoord:

Antwoord: De dataset bevat gegevens van 4324 ruimtemissies.

### Opdracht 2.5
Of een missie slaagde of faalde, wordt gegeven in de kolom met als naam `Mission_Status`. 
- Hoeveel missies slaagden en hoeveel faalden? 

In [10]:
# voorbeeldcode
# kolom met status
status = ruimtemissies["Mission_Status"]

# absolute frequentie falen en slagen
absolutefrequentietabel_status = status.value_counts() 
absolutefrequentietabel_status 

Success              3879
Failure               339
Partial Failure       102
Prelaunch Failure       4
Name: Mission_Status, dtype: int64

Antwoord:

Antwoord: 3879 missies slaagden en 445 missies faalden.

### Opdracht 2.6
Schrijf nu een Python-script dat de volgende informatie teruggeeft:
-  Het aantal succesvolle missies
-  Het aantal volledig gefaalde missies (dus de som van het aantal missies met Failure of Prelaunch Failure)
-  Het totaal aantal gefaalde missies

In [16]:
# voorbeeldcode
# falen en slagen
absolute_frequentie_slagen = absolutefrequentietabel_status["Success"]
absolute_frequentie_volledigfalen = absolutefrequentietabel_status["Failure"] + absolutefrequentietabel_status["Prelaunch Failure"]
absolute_frequentie_allefalen = aantal_ruimtemissies - absolute_frequentie_slagen 

print("Er zijn", absolute_frequentie_slagen, "geslaagde  en", absolute_frequentie_allefalen, "gefaalde ruimtemissies.")
print("Daarvan zijn er", absolute_frequentie_volledigfalen, "volledig gefaalde.")

Er zijn 3879 geslaagde  en 445 gefaalde ruimtemissies.
Daarvan zijn er 343 volledig gefaalde.


<div>
    <font color=#690027 markdown="1">
        <h3>3. Relatieve frequentie</h3> 
    </font>
</div>

### Opdracht 3.1
- Hoeveel procent van de ruimtemissies werden uitgevoerd door NASA?  

In [18]:
# voorbeeldcode
# NASA
absolute_frequentie_NASA = absolutefrequentietabel_organisaties["NASA"]
relatieve_frequentie_NASA = absolute_frequentie_NASA / aantal_ruimtemissies

print(relatieve_frequentie_NASA * 100, "procent van de ruimtemissies werd uitgevoerd door NASA.")

4.69472710453284 procent van de ruimtemissies werd uitgevoerd door NASA.


Antwoord:

Antwoord: Ongeveer 4,69 % van de ruimtemissies werd uitgevoerd door NASA.

### Opdracht 3.2
- Bereken het percentage missies dat voor de lancering is gefaald (`Prelaunch Failure`).

In [20]:
# voorbeeldcode
# Prelaunch Failure
absolute_frequentie_falenvoorlancering = absolutefrequentietabel_status["Prelaunch Failure"]
relatieve_frequentie_falenvoorlancering = absolute_frequentie_falenvoorlancering / aantal_ruimtemissies

print(relatieve_frequentie_falenvoorlancering * 100, "procent van de ruimtemissies faalde reeds voor de lancering.")

0.09250693802035154 procent van de ruimtemissies faalde reeds voor de lancering.


### Opdracht 3.3
- Stel een relatievefrequentieabel op.
- Gebruik de relatievefrequentieabel om het percentage aan ruimtemissies te achterhalen dat uitgevoerd werd door ESA.

In [22]:
# voorbeeldcode
relatievefrequentietabel_organisaties = absolutefrequentietabel_organisaties / aantal_ruimtemissies 
relatievefrequentietabel_organisaties

RVSN USSR           0.410962
Arianespace         0.064524
CASC                0.058048
General Dynamics    0.058048
NASA                0.046947
VKS RF              0.046485
US Air Force        0.037234
ULA                 0.032377
Boeing              0.031452
Martin Marietta     0.026364
SpaceX              0.023127
MHI                 0.019426
Northrop            0.019195
Lockheed            0.018270
ISRO                0.017576
Roscosmos           0.012720
ILS                 0.010638
Sea Launch          0.008326
ISAS                0.006938
Kosmotras           0.005088
US Navy             0.003932
ISA                 0.003006
Rocket Lab          0.003006
Eurockot            0.003006
ESA                 0.003006
Blue Origin         0.002775
IAI                 0.002544
ExPace              0.002313
ASI                 0.002081
CNES                0.001850
AMBA                0.001850
MITT                0.001619
JAXA                0.001619
Land Launch         0.001619
UT            

In [23]:
# voorbeeldcode
relatieve_frequentie_ESA = relatievefrequentietabel_organisaties["ESA"]
print(relatieve_frequentie_ESA * 100, "procent van de ruimtemissies werd uitgevoerd door ESA.")

0.30064754856614245 procent van de ruimtemissies werd uitgevoerd door ESA.


### Opdracht 3.4
-  Wat is het percentage missies dat een actieve `Rocket_Status` heeft?

In [25]:
# voorbeeldcode
# kolom met raketstatus
raketstatus = ruimtemissies["Rocket_Status"]

# absolute en relatieve frequentie 
absolutefrequentietabel_raketstatus = raketstatus.value_counts() 
relatievefrequentietabel_raketstatus = absolutefrequentietabel_raketstatus / aantal_ruimtemissies
relatieve_frequentie_actievestatus = relatievefrequentietabel_raketstatus["StatusActive"]

print(relatieve_frequentie_actievestatus * 100, "procent van de ruimtemissies heeft een actieve raket.")

18.270120259019425 procent van de ruimtemissies heeft een actieve raket.


<div>
    <font color=#690027 markdown="1">
        <h2>3. Oefening: Video games sales</h2> 
    </font>
</div>

Volbreng de onderstaande opdrachten gegeven een nieuwe dataset met bron https://www.kaggle.com/datasets/gregorut/videogamesales. De dataset is als csv-bestand terug te vinden in bij `data/video_games_sales.csv`

<div class="alert alert-box alert-success">
Je zal tijdens het maken van de opgaven merken dat de dateset al even niet meer is geupdated. De meest recente data is afkomstig uit 2017.
</div>

<div class="alert alert-block alert-danger"> 
Print eerst de dataset op het scherm en surf eens naar de bron van de data. Deze bevat uitleg over wat elke kolom wil zeggen. Dit helpt om een overzicht te houden tijdens de oefening. Nadien werk je per opdracht stap per stap zoals in de voorgaande paragrafen. De hele opgave in één keer proberen volbrengen verhoogt de kans op fouten.
</div>

In [28]:
videogames = pd.read_csv("data/video_games_sales.csv")
videogames

rank                                              name platform  \
0          1                                        Wii Sports      Wii   
1          2                                 Super Mario Bros.      NES   
2          3                                    Mario Kart Wii      Wii   
3          4                                 Wii Sports Resort      Wii   
4          5                          Pokemon Red/Pokemon Blue       GB   
...      ...                                               ...      ...   
16593  16596                Woody Woodpecker in Crazy Castle 5      GBA   
16594  16597                     Men in Black II: Alien Escape       GC   
16595  16598  SCORE International Baja 1000: The Official Game      PS2   
16596  16599                                        Know How 2       DS   
16597  16600                                  Spirits & Spells      GBA   

         year         genre   publisher  na_sales  eu_sales  jp_sales  \
0      2006.0        Sports    Nintendo     41.49     29.02      3.77   
1      1985.0      Platform    Nintendo     29.08      3.58      6.81   
2      2008.0        Racing    Nintendo     15.85     12.88      3.79   
3      2009.0        Sports    Nintendo     15.75     11.01      3.28   
4      1996.0  Role-Playing    Nintendo     11.27      8.89     10.22   
...       ...           ...         ...       ...       ...       ...   
16593  2002.0      Platform       Kemco      0.01      0.00      0.00   
16594  2003.0       Shooter  Infogrames      0.01      0.00      0.00   
16595  2008.0        Racing  Activision      0.00      0.00      0.00   
16596  2010.0        Puzzle    7G//AMES      0.00      0.01      0.00   
16597  2003.0      Platform     Wanadoo      0.01      0.00      0.00   

       other_sales  global_sales  
0             8.46         82.74  
1             0.77         40.24  
2             3.31         35.82  
3             2.96         33.00  
4             1.00         31.37  
...            ...           ...  
16593         0.00          0.01  
16594         0.00          0.01  
16595         0.00          0.01  
16596         0.00          0.01  
16597         0.00          0.01  

[16598 rows x 11 columns]

### Opdracht 3.1
Hoeveel videogames uit de lijst werden uitgebracht in 2015? Print de juiste waarde op het scherm.

614

### Opdracht 3.2
Genereer een relatieve frequentietabel voor de genres van elk videospel.

### Opdracht 3.3

Geef het percentage videogames uit de dataset die door Nintendo werden uitgebracht? Print de juiste waarde op het scherm in procent.

4.2354500542234

### Opdracht 3.4

Geef het percentage videogames uit de lijst die zijn uitgebracht op voor de playstation. Dus de videogames met platform PS, PS2, PS3, PS4 of PSP. Print de juiste waarde in procent op het scherm.

0.37564766839378244

### Referentielijst

Space Mission Launches. https://www.kaggle.com/datasets/sefercanapaydn/mission-launches

Video Game Sales. https://www.kaggle.com/datasets/gregorut/videogamesales

<img src="images/cclic.png" alt="Banner" align="left" width="100"/><br><br>

Notebook python in wiskunde - Statistiek, van D. De Bolster, F. wyffels & N. Gesquière in licentie gegeven volgens een <a href="http://creativecommons.org/licenses/by-nc-sa/4.0/">Creative Commons Naamsvermelding-NietCommercieel-GelijkDelen 4.0 Internationaal-licentie</a>.